src/pages/workspaces/migration/WorkspaceItem.ts

```diff
@@ -0,0 +1,115 @@
+import { div, h, span } from 'react-hyperscript-helpers';
+import { icon } from 'src/components/icons';
+import { InfoBox } from 'src/components/PopupTrigger';
+import colors from 'src/libs/colors';
+import * as Utils from 'src/libs/utils';
+import { WorkspaceMigrationInfo } from 'src/pages/workspaces/migration/migration-utils';
+
+export const WorkspaceItem = (workspaceInfo: WorkspaceMigrationInfo) => {
+  const renderMigrationState = () => {
+    const getTransferProgress = (transferType, processed, total) => {
+      if (total === 0) {
+        return `${transferType} Bucket Transfer`;
+      }
+      return `${transferType} Transfer in Progress (${Utils.formatBytes(processed)}/${Utils.formatBytes(total)})`;
+    };
+
+    const text = Utils.cond(
+      [
+        workspaceInfo.outcome === 'failure',
+        () =>
+          span({ style: { color: colors.danger() } }, [
+            'Migration Failed',
+            h(
+              InfoBox,
+              {
+                style: { marginLeft: '0.5rem' },
+                side: 'bottom',
+                tooltip: 'Failure information',
+                size: 18,
+                iconOverride: undefined,
+              },
+              [workspaceInfo.failureReason]
+            ),
+          ]),
+      ],
+      [workspaceInfo.outcome === 'success', () => span(['Migration Complete'])],
+      [workspaceInfo.migrationStep === 'ScheduledForMigration', () => span(['Starting Migration'])],
+      [workspaceInfo.migrationStep === 'PreparingTransferToTempBucket', () => span(['Preparing Original Bucket'])],
```

Comment:
I decided to avoid the word "temp" as users aren't going to understand why we need a temporary bucket.